tools/ci/templates/jobs-cpp.yaml

```diff
@@ -26,14 +26,75 @@ jobs:
+
+  # Compute the PDB package variables
+  - task: PowerShell@2
+    displayName: 'Compute PDB package variables'
+    inputs:
+      targetType: filePath
+      filePath: tools/ci/computePdbPackageVars.ps1
+    env:
+      # Read $(MRWebRTC_PdbPackageVersion) from pipeline variables
+      WRITE_VERSION: 'false'
+    condition: eq(variables['MRWebRTC_RestorePDBs'], 'true')
+
+  # Download the PDBs
+  - task: UniversalPackages@0
+    displayName: Download core PDBs
+    inputs:
+      command: download
+      vstsFeed: $(MRWebRTC_PdbFeed)
+      vstsFeedPackage: $(MRWebRTC_PdbPackageName)
+      vstsPackageVersion: $(MRWebRTC_PdbPackageVersion)
+      downloadDirectory: '$(Build.SourcesDirectory)\_pdbs'
+    condition: eq(variables['MRWebRTC_RestorePDBs'], 'true')
+
+  # Move PDBs back into their original location for the linker to find them
+  - task: PowerShell@2
+    displayName: Move core PDBs back in place
+    inputs:
+      targetType: filePath
+      filePath: tools/ci/copyPdbsForBuilding.ps1
+      arguments: '-BuildConfig ${{parameters.buildConfig}} -BuildPlatform ${{parameters.buildPlatform}} -SourcePath "$(Build.SourcesDirectory)/_pdbs" -OutputPath "bin/${{parameters.buildPlatform}}/${{parameters.buildArch}}/${{parameters.buildConfig}}"'
+    condition: eq(variables['MRWebRTC_RestorePDBs'], 'true')
+
   # Use NuGet 5.2.0
   - task: NuGetToolInstaller@1
     displayName: 'Use NuGet 5.2.0'
     inputs:
       versionSpec: 5.2.0
 
+  # Generate custom packages.config to restore only the necessary packages for the current build triple.
+  # This helps both with decreasing restore time and with minimizing disk space to avoid the 10GB limit.
+  - task: PowerShell@2
+    displayName: Generate packages.config for build triple
+    inputs:
+      targetType: filePath
+      filePath: tools/ci/generateCppPackagesConfig.ps1
+      arguments: '-BuildConfig ${{parameters.buildConfig}} -BuildPlatform ${{parameters.buildPlatform}} -BuildArch ${{parameters.buildArch}} -InputFile "$(Build.SourcesDirectory)/libs/Microsoft.MixedReality.WebRTC.Native/src/${{parameters.buildPlatform}}/packages.config" -OutputFile "bin/${{parameters.buildPlatform}}/${{parameters.buildArch}}/${{parameters.buildConfig}}/packages.config"'
+    condition: eq(variables['MRWebRTC_RestorePDBs'], 'true')
+
   # Restore the NuGet packages containing the input dependencies
   - task: 333b11bd-d341-40d9-afcf-b32d5ce6f23b@2  # NuGetCommand@2
+    condition: eq(variables['MRWebRTC_RestorePDBs'], 'true')
+    displayName: 'NuGet restore C++ lib'
+    inputs:
+      command: restore
+      restoreSolution: 'bin/${{parameters.buildPlatform}}/${{parameters.buildArch}}/${{parameters.buildConfig}}/packages.config'
+      restoreDirectory: '$(Build.SourcesDirectory)/packages'
+      includeNuGetOrg: true
+      feedsToUse: config
+      nugetConfigPath: '$(NuGetConfigPath)'
+    timeoutInMinutes: 30
+  - task: 333b11bd-d341-40d9-afcf-b32d5ce6f23b@2  # NuGetCommand@2
```

Comment:
nit: Add newline and comment, one has to re-read this a few times to see that it's a different task that executes alternatively to the previous one. 

Also the `condition` line is sometimes just below `task` and sometimes at the very end, this makes the sequence difficult to follow.